test/parallel/test-stream-pipe-await-drain-push-while-write.js

```diff
@@ -0,0 +1,29 @@
+'use strict';
+const common = require('../common');
+const stream = require('stream');
+
+// A writable stream which pushes data onto the stream which pipes into it,
+// but only the first time it's written to. Since it's not paused at this time,
+// a second write will occur. If the pipe increases awaitDrain twice, we'll
+// never get subsequent chunks because 'drain' is only emitted once.
+const writable = new stream.Writable({
+  write: common.mustCall((chunk, encoding, cb) => {
+    if (chunk.length === 32 * 1024) { // first chunk
+      readable.push(new Buffer(33 * 1024)); // above hwm
+      return process.nextTick(cb); // let pipe increment awaitDrain
```

Comment:
I wanted to test the effect (missing third buffer) rather than the cause.
In any case, `awaitDrain` isn't incremented until `_write` returns so it won't already be 2.